src/cmd_sym.ml

```diff
@@ -219,55 +213,96 @@ let run_file ~unsafe ~optimize (pc : unit Result.t Choice.t) filename =
     let*/ script = Parse.Script.from_file ~filename in
     simplify_then_link_then_run ~unsafe ~optimize pc script
 
-let get_model (solver : Thread.Solver.t) thread =
-  assert (Thread.Solver.check solver (Thread.pc thread));
-  match Thread.Solver.model solver with
-  | None -> assert false
-  | Some model -> Encoding.Model.to_string model
+let get_model solver pc =
+  assert (Solver.check solver pc);
+  match Solver.model solver with None -> assert false | Some model -> model
 
-let stop_at_first_failure = true
+let mkdir_p_exn dir =
+  let rec get_intermediate_dirs d acc =
+    if Sys.file_exists d then acc
+    else get_intermediate_dirs (Filename.dirname d) (d :: acc)
+  in
+  if Sys.file_exists dir then ()
+  else
+    let intermediate_dirs = get_intermediate_dirs dir [] in
+    List.iter (fun d -> Sys.mkdir d 0o755) intermediate_dirs
 
-let cmd profiling debug unsafe optimize workers files =
+let out_testcase ~dst ~err testcase =
+  let o = Xmlm.make_output ~nl:true ~indent:(Some 2) dst in
+  let tag ?(atts = []) name = (("", name), atts) in
+  let atts = if err then Some [ (("", "coversError"), "true") ] else None in
+  let to_string v = Format.asprintf "%a" Encoding.Value.pp_num v in
+  let input v = `El (tag "input", [ `Data (to_string v) ]) in
+  let testcase = `El (tag ?atts "testcase", testcase >>| input) in
+  let dtd =
+    "<!DOCTYPE testcase PUBLIC \"+//IDN sosy-lab.org//DTD test-format testcase \
+     1.1//EN\" \"https://sosy-lab.org/test-format/testcase-1.1.dtd\">"
```

Comment:
You can use the `{|` and `|}` delimiters to avoid the escaping:

```ocaml
let dtd =
  {|<!DOCTYPE testcase PUBLIC "+//IDN sosy-lab.org//DTD test-format testcase 1.1//EN" "https://sosy-lab.org/test-format/testcase-1.1.dtd">|}
```

Reply:
Good idea! But without escaping we're going to have a gigantic line, are you okay with that?

Reply:
I don't mind. :)